### DISCUSSION TITLE
How to read code solution

### DISCUSSION THREAD

**Discussion by abuzafarghaffari** [cssbattle/forum#398] [Q&A]
Hi,
I am new to CSS Battle and struggling to understand the solutions for previous challenges. For example, I came across a solution for the CSS Battle Daily Target on October 1, 2024, but I am having trouble figuring it out.
I don’t understand the HTML structure, and the CSS doesn’t look normal to me either. Can someone please help explain it?
Here’s the code I’m referring to:
.<style>*{background:radial-gradient(1q at 0%100%,#3F4869 63q,#E25C57 0 30vw,#3F4869)180q 476q;*{margin:150 170-15 60;scale:-1 1;box-shadow:-3cm 8q 0-56q#F4DA64

**Comment by aarthvaen** [comment]
Hi, there.

Probably a little too late for an answer, but allow me to deconstruct the code for you to potentially understand how and why it works the way it is written.

First, let's format the code back to the standard form we all know:
```
.<style>
  *
  {
    background: radial-gradient(1q at 0%100%,
#3F4869 63q,#E25C57 0 30vw,#3F4869)
180q 476q;
    
  *
    {
    margin:150 170-15 60;
    scale:-1 1;
    box-shadow:-3cm 8q 0-56q#F4DA64
    }
  }
```

At the beginning after the ```<style>``` tag, the first selector ```*``` will apply the background property for both DOM elements - ```html``` and ```body```. When it comes to code golfing, this selector is helpful for saving 3 characters of your solution.

Next up, we have the background property which is definited by a radial gradient:

```
radial-gradient(1q at 0%100%,
/* the 1q at the beginning is a replacement 
for the default parameter "circle". 
The "at 0%100%" parameter sets the 
starting point of our circle to bottom 
left of the canvas.*/

#3F4869 63q,#E25C57 0 30vw,#3F4869)
/* Next, we set up the inner and outer circle.
The inner circle will have the main required 
background color and a size of 63q (~60px). 
The outer circle has this "0" parameter before size
declaration, which in radial-gradient's default syntax,
sets the blurring effect between these two circles.
Finally, we set the size of the outer circle to 
30vw (120px) and we end the radial-gradient 
setup with the same background color 
we used at the beginning.

180q 476q
/* These two values are the shorthand for
"background-position". Since gradients belong 
to the <image> data type in CSS, you can pretty
 much manipulate gradients the same way you
 manipulate actual images for your 
website. In this case, we move the
radial-gradient by 180q (~170px) horizontally, 
and 476q (~450px) vertically.
```

As you might've wrote this code whilst learning how it works, you may notice a weird rendering, that doesn't click with the original target. 

This is because you'll have the same background output twice and, in order to match our target, we need some 
additional properties that will only be applied on ```BODY``` element:

```
    margin:150 170-15 60;
/* This margin property will set the size of
the ```body``` element, so that the duplicate
circle quarter in our rendering will placed on the
left side. Note that the radial-gradient will 
recculate the output for your body element
in relation with the defined margin values.*/

    scale:-1 1;
/* This property will flip the second circle quarter
 on the horizontal axis so it matches 
with our target.*/

    box-shadow:-3cm 8q 0-56q#F4DA64
/* This box-shadow will create the 
needed yellow shadow at the 
bottom right corner. 
It's set at -3cm on x axis, 8q on y axis, 
no blur and -56q off the box's original size. */

Finally, the dot at the very beginning of the code helps the top right circle quarter to stay visible. It's a trick tjst has been used multiple times in daily targets, but I can assume someone will explain it better than me.

Hope this all helps in a way or another.

**Reply by ZeroOne00001** [reply]
So just one simple question is this a valid answer , for the output , because i am not seeing anything

<style>*{background:#191210;color:84271C;border-radius:1in;margin:0 65%225-265;box-shadow:25vw 75vh,-31q 231q,-60q 2in,425px 0,590q 9q,615q 33q;*{margin:-125 165

this is the top solution for oct 26

**Reply by SelenIT** [reply]
@ZeroOne00001 it displays correctly in Chromium-based browsers. You probably test it in Firefox or other browser that doesn't stretch the height of `html` and `body` elements by default. The CSSBattle backend uses Chrome to evaluate the solutions, so many solutions rely on Chromium-specific behaviors like this and it's recommended to use Chromium-based browsers for the game
